### PR TITLE
116 add comment existing eng

### PR DESCRIPTION
**Pull request by bcloutier7**
# Description

[#116 Add comment to existing engagement (client side only)]
https://taiga.dts-stn.com/project/pond-knowlege-management-portal/us/116?milestone=20

Added capability to add comments for an engagement. (only client side)

# Story Acceptance Criteria

Respecting both AC in taiga. 

Get the acceptance criteria for the story relating to your task and put all of it here.

## Test Instructions

1. Select an engagement.
2. scroll down to the comment section
3. add a comment and see it added to the list

## Help Requested

The box to view all comments might be a bit small. Let me know if I should make it bigger.
(User and date are hardcoded strings)

## Checklist
- [ ] Strings use placeholders for translation (No hard coded strings)
- [ ] Unit tests have been added/updated

### CODE REVIEW

File: components/engagement/EngComments.vue [hunk 1]
```diff
@@ -1,70 +1,48 @@
...
   data() {
     return {
-      commentInfo: {
-        eng_id: '5ef4ee00a7cc2a950e1d5641',
-        content: 'Testing new comments '
-      }
+      // commentInfo: {
+      //   eng_id: '5ef4ee00a7cc2a950e1d5641',
+      //   content: ''
+      // }
```
Comment (andr0272) [inline components/engagement/EngComments.vue:44]: Remove this please.

File: components/engagement/EngComments.vue [hunk 2]
```diff
@@ -73,43 +51,58 @@ export default {
...
 
   methods: {
-    goBack() {
-      this.$router.back()
-    },
-
-    submitForm(commentInfo) {
-      this.$store.dispatch('engagements/addComment', commentInfo)
+    submitComment() {
+      if (this.newComment !== '') {
```
Comment (andr0272) [inline components/engagement/EngComments.vue:55]: Just use this.newComment. Check out [this](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) JS concept

**Comment by bcloutier7** [comment]
Thank you for this clarification Jordan.

File: pages/view/engagement/_id.vue [hunk 3]
```diff
@@ -56,12 +56,15 @@
...
+          <EngComments
+            :comments="engagement.comments"
+            @submit-comment="submitComment"
+          />
+          <!-- <ul>
             <li v-for="(comment, index) in engagement.comments" :key="index">
               {{ comment }}
             </li>
-          </ul>
+          </ul> -->
```
Comment (andr0272) [inline pages/view/engagement/_id.vue:67]: Remove this

File: pages/view/engagement/_id.vue [hunk 4]
```diff
@@ -138,6 +143,15 @@ export default {
...
     goBack() {
       this.$router.back()
+    },
+    submitComment(newComment) {
+      // this.engagement.comments.push({
+      //   // id: this.comments.length + 1, generate new id here
+      //   // date: new Date(),
+      //   // content: newComment
+      //   newComment
+      // })
```
Comment (andr0272) [inline pages/view/engagement/_id.vue:153]: Remove this

**Comment by will0684** [comment]
> image
> I understand that validation isn't required for this task but could the comment wrap if it hits a certain length? The way its implemented right now stretches my screen 😩
> 
> A few small changes, but other than that, very impressive implementation 💯

File: components/engagement/EngComments.vue (hunk 1 above)
Comment (bcloutier7) [reply components/engagement/EngComments.vue:44]: On it! Thank you for that catch Adam

**Pull request merged**